src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -83,16 +83,54 @@ public static bool IsResumableChangeStreamException(Exception exception)
 
         public static bool IsRetryableWriteException(Exception exception)
         {
+            if (__retryableWriteExceptions.Contains(exception.GetType()))
+            {
+                return true;
+            }
+
             var commandException = exception as MongoCommandException;
             if (commandException != null)
             {
                 var code = (ServerErrorCode)commandException.Code;
-                return __retryableWriteErrorCodes.Contains(code);
+                if (__retryableWriteErrorCodes.Contains(code))
+                {
+                    return true;
+                }
             }
-            else
+
+            var writeConcernException = exception as MongoWriteConcernException;
+            if (writeConcernException != null)
             {
-                return __retryableWriteExceptions.Contains(exception.GetType());
+                var writeConcernError = writeConcernException.WriteConcernResult.Response.GetValue("writeConcernError", null)?.AsBsonDocument;
+                if (writeConcernError != null)
+                {
+                    var code = (ServerErrorCode)writeConcernError.GetValue("code", -1).AsInt32;
+                    switch (code)
+                    {
+                        case ServerErrorCode.InterruptedAtShutdown:
```

Comment:
This will eventually be the same list as is used for retryable write error codes above, according to https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#terms ?

Are you waiting to merge the new retryable writes tests before making that change, or am I misunderstanding something?

Reply:
I haven't merged any new retryable writes tests.

This PR is about transaction tests only. Is that OK?

Reply:
Yes, that's fine.